File: win-prog/ProgWinCS/String Theory/StringWriterAppend/StringWriterAppend.cs
//-------------------------------------------------
// StringWriterAppend.cs © 2001 by Charles Petzold
//-------------------------------------------------
using System;
using System.IO;

class StringWriterAppend
{
	const int iIterations = 10000;

	public static void Main()
	{
		DateTime     dt = DateTime.Now;
		StringWriter sw = new StringWriter();

		for (int i = 0; i < iIterations; i++)
			sw.WriteLine("abcdefghijklmnopqrstuvwxyz");

		string str = sw.ToString();

		Console.WriteLine(DateTime.Now - dt);
	}
}

File: win-prog/ProgWinVB/String Theory/StringWriterAppend/StringWriterAppend.vb
'---------------------------------------------------
' StringWriterAppend.vb (c) 2002 by Charles Petzold
'---------------------------------------------------
Imports System
Imports System.IO

Module StringWriterAppend
    Const iIterations As Integer = 10000

    Sub Main()
        Dim dt As DateTime = DateTime.Now
        Dim sw As New StringWriter()
        Dim i As Integer

        For i = 1 To iIterations
            sw.WriteLine("abcdefghijklmnopqrstuvwxyz" & vbCrLf)
        Next i
        Dim str As String = sw.ToString()

        Console.WriteLine(DateTime.op_Subtraction(DateTime.Now, dt))
    End Sub
End Module
